### ISSUE TITLE
Rule Change: report `super()` if it is accessed after `this`

### ISSUE THREAD

**Issue by Tanujkanti4441** [eslint/eslint#21016]
### What rule do you want to change?

`constructor-super`

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
class A extends B {
  constructor(name) {
	this.name = name;  
	super();
  }
}
```

### What does the rule currently do for this code?

Rule does not report this code, even though it gives an error because `super()` must be called before `this` keyword.

`ReferenceError: Must call super constructor in derived class before accessing 'this' or returning from derived constructor`.

https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgY29uc3RydWN0b3Itc3VwZXI6IFwiZXJyb3JcIiovXG5cbmNsYXNzIEEgZXh0ZW5kcyBCIHtcbiAgY29uc3RydWN0b3IobmFtZSkge1xuXHR0aGlzLm5hbWUgPSBuYW1lOyAgXG5cdHN1cGVyKCk7XG4gIH1cbn0iLCJvcHRpb25zIjp7InJ1bGVzIjp7fSwibGFuZ3VhZ2VPcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=

### What will the rule do after it's changed?

Rule will report the super() if it is accessed after `this` in the constructor.

### Participation

- [x] I am willing to submit a pull request to implement this change.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

**Comment by kirkwaiblinger** [comment]
https://eslint.org/docs/latest/rules/no-this-before-super
